src/commands/app/upgrade.tsx

```diff
@@ -0,0 +1,230 @@
+import { ExecRenderBaseCommand } from "../../rendering/react/ExecRenderBaseCommand.js";
+import {
+  appInstallationArgs,
+  withAppInstallationId,
+} from "../../lib/app/flags.js";
+import { projectFlags } from "../../lib/project/flags.js";
+import { Flags, ux } from "@oclif/core";
+import React, { ReactNode } from "react";
+import { Text } from "ink";
+import {
+  getAppFromUuid,
+  getAppInstallationFromUuid,
+  getAppVersionFromUuid,
+} from "../../lib/app/uuid.js";
+import {
+  getAllUpgradeCandidatesFromAppInstallationId,
+  getAvailableTargetAppVersionFromExternalVersion,
+  getLatestAvailableTargetAppVersionForAppVersionUpgradeCandidates,
+} from "../../lib/app/versions.js";
+import {
+  makeProcessRenderer,
+  ProcessFlags,
+  processFlags,
+} from "../../rendering/process/process_flags.js";
+import { Success } from "../../rendering/react/components/Success.js";
+import { ProcessRenderer } from "../../rendering/process/process.js";
+import { MittwaldAPIV2, MittwaldAPIV2Client } from "@mittwald/api-client";
+import { waitUntilAppIsUpgraded } from "../../lib/app/wait.js";
+import { assertStatus } from "@mittwald/api-client-commons";
+
+type AppApp = MittwaldAPIV2.Components.Schemas.AppApp;
+type AppAppInstallation = MittwaldAPIV2.Components.Schemas.AppAppInstallation;
+type AppAppVersion = MittwaldAPIV2.Components.Schemas.AppAppVersion;
+
+export class UpgradeApp extends ExecRenderBaseCommand<typeof UpgradeApp, void> {
+  static description = "Upgrade target appinstallation to target version";
+  static args = {
+    ...appInstallationArgs,
+  };
+  static flags = {
+    "target-version": Flags.string({
+      description: "Target version to upgrade target app to.",
+    }),
+    wait: Flags.boolean({
+      description: "wait for the upgrade process to finish",
+      char: "w",
+    }),
+    force: Flags.boolean({
+      char: "f",
+      description: "Do not ask for confirmation.",
+    }),
+    ...projectFlags,
+    ...processFlags,
+  };
+
+  protected async exec(): Promise<void> {
+    const process = makeProcessRenderer(
+      this.flags as ProcessFlags,
+      "App upgrade",
+    );
+    const appInstallationId: string = await withAppInstallationId(
+        this.apiClient,
+        UpgradeApp,
+        this.flags,
+        this.args,
+        this.config,
+      ),
+      currentAppInstallation: AppAppInstallation =
+        await getAppInstallationFromUuid(this.apiClient, appInstallationId),
+      currentApp: AppApp = await getAppFromUuid(
+        this.apiClient,
+        currentAppInstallation.appId as string,
+      ),
+      currentAppVersion: AppAppVersion = await getAppVersionFromUuid(
+        this.apiClient,
+        currentApp.id,
+        (currentAppInstallation.appVersion as { current: string }).current,
```

Comment:
Why is the `as { current: string }` typecast necessary?

Reply:
As above, this is a workaround since I didnt understand that its just a missing check if it even is undefined.
Fixed with a check and exit with error code 1 if its undefined.